File: java/String-2/endOther.java
//Given two strings, return true if either of the strings appears at the very end of the other string, ignoring upper/lower case differences (in other words, the computation should not be "case sensitive"). Note: str.toLowerCase() returns the lowercase version of a string.
//
//endOther("Hiabc", "abc") → true
//endOther("AbC", "HiaBc") → true
//endOther("abc", "abXabc") → true

public boolean endOther(String a, String b) {
  int alen= a.length();
  int blen=b.length();
  if (alen==0 || blen ==0)
    return true;
  if (blen>=alen && b.substring(blen-alen).toLowerCase().equals(a.toLowerCase()))
    return true;
  if (alen>blen && a.substring(alen-blen).toLowerCase().equals(b.toLowerCase()))
    return true;
  return false;
}

File: python/String-2/end_other.py
#Given two strings, return True if either of the strings appears at the very end of the other string, ignoring upper/lower case differences (in other words, the computation should not be "case sensitive"). Note: s.lower() returns the lowercase version of a string.
#
#end_other('Hiabc', 'abc') → True
#end_other('AbC', 'HiaBc') → True
#end_other('abc', 'abXabc') → True

def end_other(a, b):
  if (len(a)==0 or len(b)==0):
    return True
  if (len(a)>=len(b)):
    if(a[0-len(b):].lower()==b.lower()):
      return True
  if(b[0-len(a):].lower()==a.lower()):
    return True
  return False
